### ISSUE TITLE
Support dot notation, `$elemMatch` and error handling of projection positional `$` operator

### ISSUE THREAD

**Issue by chilagrow** [FerretDB/FerretDB#2522]
### What should be done?

Blocked by https://github.com/FerretDB/FerretDB/issues/1709

In https://github.com/FerretDB/FerretDB/issues/1709, basic implementation of projection operator is done. This issue is about ensuring that dot notation and operator works and handling unhappy path.

https://www.mongodb.com/docs/manual/reference/operator/projection/positional/#mongodb-projection-proj.-

DoD:
Happy path:
- Use dot notation for array path e.g. `db.values.find({"v.foo.bar": {$gt: "a"}}, {"v.foo.$": 1})`
- Use `$elemMatch` to match array field e.g. `db.values.find({v: {$elemMatch: {$gt: 42}}}, {"v.$": 1})`

Unhappy path:
- One positional operator is allowed e.g. `db.values.find({v: {$gt: 42}}, {"v.$.foo.$": 1})`
- Positional operator cannot be on its own e.g. `db.values.find({v: {$gt: 42}}, {"$": 1})`
- Exclusion of positional operator e.g. `db.values.find({v: {$gt: 42}}, {"v.$": 0})`
- Other negative test cases

Above examples are tested with task env-data and should work for our [shareddata](https://github.com/FerretDB/FerretDB/tree/main/integration/shareddata)

Documentation:

update supported commands doc
create a new issue to write documentation about this feature (for the techwriter)

### Where?

https://github.com/FerretDB/FerretDB/blob/main/internal/handlers/common/projection.go

### Definition of Done

- all handlers updated;
- integration/compatibility tests added/updated;
- spot refactorings done;
- user documentation updated or an issue to create documentation created;
